domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -245,72 +243,76 @@ class TopicListController @Inject constructor(
     topicProgressList: List<TopicProgress>,
     completionTimeFilter: (Long) -> Boolean
   ): List<PromotedStory> {
+    if (topicProgressList.isNotEmpty()) {
```

Comment:
Why is this check needed? Won't an empty list still be returned even without it?

I'm also slightly confused about the logic here. It seems exactly the same except one change: before we didn't set the promoted story list at all if there wasn't topic progress, but now we set an empty list. How does that result in the correct recommendations being shown?

Reply:
Before I had this check in `computePromotedActivityList()` if it is empty it was returning `promotedActivityListBuilder` with empty list. But as we need to initially recommend some topics without any progress I moved this check to `computePlayedStories()`. Here we doing some computation on topic's that are ongoing so we need to check if it not empty.

Reply:
I thought if I don't add a check it would result in crash. I cross checked and if I don't add check it works fine. So removed this check and updated code.

Reply:
The new implementation seems resilient to crashing since it's always operating on the original list (which is empty).

What I don't understand is: why does empty topic progress list result in the first two topics of the graph being suggested? That's how it's supposed to work (per computeSuggestedStories), but why does a now-empty promoted story list trigger computeSuggestedStories to return the correct thing?

Reply:
As per earlier documentation, unless user has any ongoing topic it wasn't suggesting any topics, which in this case gives recommendation for selected topics. Our previous logic was all correct except that, we were not entering into computation logic of suggestion, if there are no topics in progress. Hope this clears your doubt.

Reply:
Ah, I see where I was making the mistake in reading the code. This does make complete sense now--thank you. :)